simpeg/directives/directives.py

```diff
@@ -140,7 +142,7 @@ def opt(self):
         return self.invProb.opt
 
     @property
-    def reg(self):
+    def reg(self) -> ComboObjectiveFunction:
```

Comment:
Since the regularizations can either be a combo or an objective function, I think we should update these type hints:
```suggestion
    def reg(self) -> BaseObjectiveFunction:
```